rfcs/017-fair-share-scheduling.md

```diff
@@ -0,0 +1,216 @@
+---
+name: Formal RFC
+about: Submit a formal Request For Comments for consideration by the team.
+title: 'Fair Share Job Scheduling for CouchDB 3.x Replicator'
+labels: rfc, discussion
+assignees: 'vatamane@apache.org'
+
+---
+
+# Introduction
+
+This document describes an improvement to the CouchDB 3.x replicator to
```

Comment:
I think this PR should go to `3.x` and then the `3.x` reference should be omitted from the RFC.

Reply:
It's specifically targeted to towards the 3.x scheduler which is architecturally different from the `main` one.  The 3.x branch is actually easier to apply these changes to so it seemed like a good place to start. For the main replicator we might have a separate RFC or just update this one with a PR.

As discussed below we don't have a 3.x branch yet, if and when do we could move it there.

Reply:
> It's specifically targeted towards the 3.x scheduler which is architecturally different from the main one. 

Yeah, I got that part. I just thought as `HEAD` here refers to `3.x`-related content, `3.x` in the change would just be redundant / confusing. Do you expect this RFC to be included in the `main` branch as well? (A yes and a no answer would both be reasonable to me.)

Reply:
However most of the RFCs we have are actually for 4.x, so it's better to be explicit probably. Especially if for this case there will be another RFC targeting 4.x (unless we just update this one and highlight differences).

Reply:
Yeah, I was hoping for the "we just update this one and highlight differences".